### PR TITLE
Fature/rebranding and signin

### PR DESCRIPTION
**Pull request by iceraluk**
- Rebranding from PMP to Tailspend
- removed redundant create_user_if_needed call
- Added two Sign in buttons for the two suppliers
- Added ALLOW_USER_PASSWORD_AUTH explicit_auth_flows to be checked as newly created client apps do not have the USER_PASSWORD_AUTH auth flow

### CODE REVIEW

**Comment by iceraluk** [comment]
@tim-s-ccs if you get the chance I would appreciate a review on this. thank you!

**Comment by tim-s-ccs** [comment]
> @tim-s-ccs if you get the chance I would appreciate a review on this. thank you!

I'll happily take a look when I get a moment 😄

**Review by tim-s-ccs — APPROVED** [review]
Downloaded and tried it out an everything looks good 😃 

The only thing I can think of is stuff in the README that might relate to PMP Keycloak that might need to be updated?

File: app/services/cognito/sign_in_user.rb [hunk 1]
```diff
@@ -51,8 +51,7 @@ def success?
     def initiate_auth
       cognito_common = Cognito::Common.new
       client_creds = cognito_common.get_client_credentials(client_id)
-
-      if client_creds.user_pool_client.explicit_auth_flows.include? 'USER_PASSWORD_AUTH'
+      if (client_creds.user_pool_client.explicit_auth_flows & %w[USER_PASSWORD_AUTH ALLOW_USER_PASSWORD_AUTH]).any?
```
Comment (tim-s-ccs) [inline app/services/cognito/sign_in_user.rb:54]: Why did you need to do this?

Comment (iceraluk) [reply app/services/cognito/sign_in_user.rb:54]: it looks like the new app clients do not return USER_PASSWORD_AUTH explicit auth flow anymore, but they return ALLOW_USER_PASSWORD_AUTH instead for the 'Enable username-password (non-SRP) flow'

**Comment by iceraluk** [comment]
Good spot, I have made some changes to the Readme file. The keycloak setup still stands, I have just changed it to a docker image that works. Also have changed the PRINT_MAKERPLACE_CALLBACK_URL environment variable with the new ones for the two sign in buttons.

**Comment by iceraluk** [comment]
> Downloaded and tried it out an everything looks good 😃
> 
> The only thing I can think of is stuff in the README that might relate to PMP Keycloak that might need to be updated?

Good spot, I have made some changes to the Readme file. The keycloak setup still stands, I have just changed it to a docker image that works. Also have changed the PRINT_MAKERPLACE_CALLBACK_URL environment variable with the new ones for the two sign in buttons.
